external/src/main/java/com/redhat/cloud/policies/engine/process/EventProcessor.java

```diff
@@ -0,0 +1,127 @@
+package com.redhat.cloud.policies.engine.process;
+
+import com.redhat.cloud.policies.api.model.condition.expression.ExprParser;
+import com.redhat.cloud.policies.engine.db.repositories.PoliciesRepository;
+import com.redhat.cloud.policies.engine.db.entities.Policy;
+import com.redhat.cloud.policies.engine.db.repositories.PoliciesHistoryRepository;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.annotation.Metric;
+import org.hawkular.alerts.api.model.event.Event;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static com.redhat.cloud.policies.engine.process.PayloadParser.CHECK_IN_FIELD;
+import static com.redhat.cloud.policies.engine.process.PayloadParser.DISPLAY_NAME_FIELD;
+import static com.redhat.cloud.policies.engine.process.PayloadParser.INVENTORY_ID_FIELD;
+import static java.time.ZoneOffset.UTC;
+
+@ApplicationScoped
+public class EventProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(EventProcessor.class);
+    private static final Pattern ACTIONS_SPLIT_PATTERN = Pattern.compile(";");
+    private static final List<String> NOTIFICATIONS_ACTIONS = List.of("email", "notification");
```

Comment:
I guess this could be simplified to only notifications (gabi might confirm). At some point we did migrate the SQL from `email` to `notification`- This was added because we didn't dare to migrate the infinispan data :grin:

Reply:
It could indeed. I'll check with gabi first when I'll have access to the policies DB.